apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/purchase-notice/index.jsx

```diff
@@ -0,0 +1,34 @@
+import { useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useEffect, useState } from 'react';
+import './style.scss';
+
+function PurchaseNotice() {
+	const [ hasPaymentNotice, setHasPaymentNotice ] = useState( false );
+	const { createNotice } = useDispatch( noticesStore );
+
+	useEffect( () => {
```

Comment:
A late review: if you call `useEffect()` with no dependency array, it will be run on every render, which is more or less the same as not using `useEffect()` at all. I guess that was the point of using `hasPaymentNotice`, but still, calling this block on every render is not necessary and I suspect was accidental.

If you want to run the `useEffect()` only once when the component first mounts, use an empty dependency array.

Reply:
Addressed by #66138